.github/workflows/post-atlas-operator-release.yaml

```
@@ -55,6 +55,15 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Create and push branch
         run: |
+          export VERSION=${{ github.event.inputs.version }}
+          export MAJOR_VERSION=$(cat operator/mongodb-atlas-kubernetes/major-version)
+          ## Check that VERSION starts with MAJOR_VERSION
+          if [[ ${VERSION} == ${MAJOR_VERSION}.* ]]; then
+            echo "Version OK";
+          else
+            echo "Bad major version for ${VERSION}, expected ${MAJOR_VERSION}.x.y";
```

given we don't have a proper stack trace in scripts environments an output would be like:
```
Bad major version for wronginput, expected 2.x.y"
```
This can be confusing to read. Suggestion:
```
Version mismatch: target version "wronginput" doesn't match the Kubernetes Atlas Operator major version "2.x.y". See https://github.com/mongodb/mongodb-atlas-kubernetes/blob/main/docs/dev/release.md for detailed instructions.
```

---

SGTM

```suggestion
            echo "Version mismatch: target version ${VERSION} doesn't match the Kubernetes Atlas Operator major version ${MAJOR_VERSION}.x.y";
```

This message is for us, and there is no specific help on our release on this point I think, So I would skip the link for now.